src/linear_algebra/matrix.lean

```diff
@@ -1029,3 +1031,144 @@ lemma matrix.dot_product_eq_zero_iff {v : n → R} : (∀ w, matrix.dot_product
 ⟨λ h, matrix.dot_product_eq_zero v h, λ h w, h.symm ▸ zero_dot_product w⟩
 
 end
+
+namespace matrix
+
+variables {m n : Type u} [decidable_eq n] [fintype n] [decidable_eq m] [fintype m] {R : Type v}
+  [comm_ring R]
+
+lemma to_block_matrix_det (M : matrix m m R) (p : m → Prop) [decidable_pred p] :
+  M.det = (matrix.from_blocks (to_block M p p) (to_block M p (λ j, ¬p j))
+    (to_block M (λ j, ¬p j) p) (to_block M (λ j, ¬p j) (λ j, ¬p j))).det :=
+begin
+  rw ← matrix.det_reindex_self (equiv.sum_compl p).symm M,
+  unfold det,
+  congr, ext σ, congr, ext,
+  generalize hy : σ x = y,
+  cases x; cases y;
+  simp only [matrix.reindex_apply, to_block_apply, equiv.symm_symm,
+    equiv.sum_compl_apply_inr, equiv.sum_compl_apply_inl,
+    from_blocks_apply₁₁, from_blocks_apply₁₂, from_blocks_apply₂₁, from_blocks_apply₂₂],
+end
+
+lemma to_square_block_det (M : matrix m m R) {n : nat} (b : m → fin n) (k : fin n) :
+  (to_square_block M b k).det = (to_square_block_prop M (λ i, b i = k)).det := by simp
```

Comment:
There is a lot going on in this line, so let's make the proof a bit more obvious to find.
```suggestion
  (to_square_block M b k).det = (to_square_block_prop M (λ i, b i = k)).det :=
by simp
```